x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -350,67 +355,80 @@ private void transferDataToEnrichIndex(final String destinationIndexName) {
         reindexRequest.getDestination().source(new BytesArray(new byte[0]), XContentType.SMILE);
         reindexRequest.getDestination().routing("discard");
         reindexRequest.getDestination().setPipeline(EnrichPolicyReindexPipeline.pipelineName());
-        client.execute(ReindexAction.INSTANCE, reindexRequest, new ActionListener<>() {
-            @Override
-            public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
-                // Do we want to fail the request if there were failures during the reindex process?
-                if (bulkByScrollResponse.getBulkFailures().size() > 0) {
-                    logger.warn(
-                        "Policy [{}]: encountered [{}] bulk failures. Turn on DEBUG logging for details.",
-                        policyName,
-                        bulkByScrollResponse.getBulkFailures().size()
-                    );
-                    if (logger.isDebugEnabled()) {
-                        for (BulkItemResponse.Failure failure : bulkByScrollResponse.getBulkFailures()) {
-                            logger.debug(
-                                new ParameterizedMessage(
-                                    "Policy [{}]: bulk index failed for index [{}], id [{}]",
-                                    policyName,
-                                    failure.getIndex(),
-                                    failure.getId()
-                                ),
-                                failure.getCause()
+
+        // The ContextPreservingActionListener here is for the purpose of dropping the response headers, as we need this reindex to run
+        // in the security context of the user (rather than Enrich's security context) to ensure that DLS/FLS is correctly applied, but
+        // the reindex needs to access the `.enrich` index, which causes a deprecation warning. Since we drop the response headers,
+        // the deprecation warning is also dropped - but this is a hack and will not work once full protections of system indices are
```

Comment:
Do we have an issue for this yet?

Reply:
We do now: https://github.com/elastic/elasticsearch/issues/62505